simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp

```diff
@@ -383,11 +383,19 @@ void SimplePlanningSimulator::on_engage(const Engage::ConstSharedPtr msg)
 }
 
 void SimplePlanningSimulator::on_control_mode_request(
-  const ControlModeRequest::Request::SharedPtr request,
-  const ControlModeRequest::Response::SharedPtr response)
-{
-  current_control_mode_ = request->mode;
-  response->success = true;
+  const ControlModeCommand::Request::SharedPtr request,
+  const ControlModeCommand::Response::SharedPtr response)
+{
+  const auto m = request->mode;
+  if (m == ControlModeCommand::Request::MANUAL) {
+    current_control_mode_.mode = ControlModeReport::MANUAL;
+    response->success = true;
+  } else if (m == ControlModeCommand::Request::AUTONOMOUS) {
+    current_control_mode_.mode = ControlModeReport::AUTONOMOUS;
+    response->success = true;
+  } else {  // not supported
+    response->success = false;
```

Comment:
Adding error messages might make the node user friendly.
e.g.,
```
RCLCPP_ERROR(this->get_logger(), "Requested mode not supported");
```

Reply:
done